src/Acl/Policies/CharacterPolicy.php

```diff
@@ -107,6 +109,31 @@ private function isCeo(User $user, CharacterInfo $character)
         return in_array($corporation->ceo_id, $user->associatedCharacterIds());
     }
 
+    /**
+     * @param \Seat\Web\Models\User $user
+     * @param \Seat\Eveapi\Models\Character\CharacterInfo $character
+     * @return bool
+     */
+    private function isDirector(User $user, CharacterInfo $character): bool
+    {
+        // retrieve corporation to which the character is assigned
+        $corporation_id = $character->affiliation->corporation_id ?? null;
+
+        // in case we were not able to find the corporation ID - assume the user is not Director of their corporation
+        if (is_null($corporation_id)) {
+            return false;
+        }
+
+        // attempt to retrieve roles for any associated character related to the corporation
+        $corporationRoles = CorporationRole::where('corporation_id', $corporation_id)
+            ->whereIn('character_id', $user->associatedCharacterIds())
+            ->where('role', 'Director')
+            ->where('type', 'roles')->get();
```

Comment:
why do you need to get here ? you just want to know if you have a match. You can use `exists`